src/plone/restapi/deserializer/dxfields.py

```diff
@@ -97,12 +101,17 @@ def __call__(self, value):
         # objects.
         # We try to guess the correct deserialization from the current field
         # value.
-        dm = queryMultiAdapter((self.context, self.field), IDataManager)
-        current = dm.get()
-        if current is not None:
-            tzinfo = current.tzinfo
+        is_publication_field = self.field.interface == IPublication
+        if is_publication_field:
+            # because IPublication datamanager strips timezones
+            tzinfo = pytz.timezone(default_timezone())
```

Comment:
@cekk While this is way better than assuming UTC always, it can still have issues if the server has different timezone than the one from registry, **UTC** for example (the default behavior of `plone/plone-backend` Docker image).

And I think @sneridagh can confirm that their installations doesn't provide the TZ environment variable to docker containers, thus **UTC** by default. While the default timezone registry entry when creating a new Plone instance is `Europe/Berlin`.

How to test:
```
docker run -it --rm -p 8080:8080 -e SITE=Plone plone/plone-backend bash

git clone https://github.com/plone/plone.restapi.git
cd plone.restapi
git checkout handle_tz_in_publication_fields
cd /app
bin/pip install -e ./plone.restapi --no-deps
./docker-entrypoint.sh start
```

Now add a News Item with Effective Date via Volto and check portal_catalog entry.